File: SCALA/Cap08/c08ex05.scala
// c08ex05.scala

import scala.io.StdIn._

object Main {
  def main(args: Array[String]): Unit = {
    var NOME = new Array[String](10)

    var RESP = "SIM"
    var PESQ: String = ""
    var RESPOSTA: String = ""
    var I: Int = 0
    var ACHA: Boolean = false

    println("PESQUISA SEQUENCIAL DE NOMES\n")

    for (I <- 0 to 9) {
      printf("Entre o %2do. nome: ", I + 1)
      NOME(I) = readLine()
    }

    while (RESP.equals("SIM")) {
      print("\nEntre o nome a ser pesquisado: ")
      PESQ = readLine()
      I = 0
      ACHA = false

      while (I <= 9 && ACHA == false) {
        if (PESQ.equals(NOME(I)))
          ACHA = true
        else
          I += 1
      }

      println()
      if (ACHA == true)
        println(s"$PESQ foi localizado na posicao ${ I + 1 }")
      else
        println(s"$PESQ nao foi localizado")

      do {
        print("\nDeseja continuar? (SIM/NAO): ")
        RESPOSTA = readLine()
        RESP = RESPOSTA.toUpperCase()
        if (!RESP.equals("SIM") && !RESP.equals("NAO"))
          println("Informe apenas SIM ou NAO.")
      } while (!RESP.equals("SIM") && !RESP.equals("NAO"))
    }

    print("\nTecle <Enter> para encerrar... ")
    readLine()
  }
}

File: DART/Cap08/c08ex05.dart
// c08ex05.dart

import 'dart:io';

void main() {
  List<String> NOME = List.filled(10, "");
  String RESP = "SIM";
  String PESQ = "";
  String RESPOSTA = "";
  bool ACHA = false;

  print("PESQUISA SEQUENCIAL DE NOMES\n");

  for (int I = 0; I <= 9; I++) {
    stdout.write("Entre o ${(I + 1).toString().padLeft(2)}o. nome: ");
    NOME[I] = stdin.readLineSync() ?? "";
  }

  while (RESP == "SIM") {
    stdout.write("\nEntre o nome a ser pesquisado: ");
    PESQ = stdin.readLineSync() ?? "";
    int I = 0;
    ACHA = false;

    while (I <= 9 && ACHA == false) {
      if (PESQ == NOME[I])
        ACHA = true;
      else
        I++;
    }

    print("");
    if (ACHA == true)
      print("$PESQ foi localizado na posicao ${I + 1}");
    else
      print("$PESQ nao foi localizado");

    do {
      stdout.write("\nDeseja continuar? (SIM/NAO): ");
      RESPOSTA = stdin.readLineSync() ?? "";
      RESP = RESPOSTA.toUpperCase();
      if (RESP != "SIM" && RESP != "NAO")
        print("Informe apenas SIM ou NAO.");
    } while (RESP != "SIM" && RESP != "NAO");
  }

  stdout.write("\nTecle <Enter> para encerrar... ");
  stdin.readLineSync();
}
